src/middlewared/middlewared/plugins/filesystem.py

```diff
@@ -9,19 +9,27 @@
 import pathlib
 
 from middlewared.event import EventSource
-from middlewared.schema import accepts, Bool, Dict, Float, Int, List, Ref, returns, Path, Str
-from middlewared.service import private, CallError, filterable_returns, Service, job
-from middlewared.utils import filter_list
 from middlewared.plugins.pwenc import PWENC_FILE_SECRET
 from middlewared.plugins.cluster_linux.utils import CTDBConfig, FuseConfig
 from middlewared.plugins.filesystem_ import stat_x
+from middlewared.schema import accepts, Bool, Dict, Float, Int, List, Ref, returns, Path, Str
+from middlewared.service import private, CallError, filterable_returns, Service, job
+from middlewared.utils import filter_list, run
 
 
 class FilesystemService(Service):
 
     class Config:
         cli_namespace = 'storage.filesystem'
 
+    @private
+    async def update_immutable_flag_of_path(self, set_flag, path):
```

Comment:
`set` is shorter than `update` and `path` is redundant as we are already in filesystem plugin. `set_immutable` can be a better name.

This operation seems to be performed by `FS_IOC_SETFLAGS` ioctl, can you investigate it instead of using subprocess?